python/cudf/cudf/core/dataframe.py

```diff
@@ -842,6 +847,21 @@ def _init_from_list_like(self, data, index=None, columns=None):
         elif len(data) > 0 and isinstance(data[0], pd._libs.interval.Interval):
             data = DataFrame.from_pandas(pd.DataFrame(data))
             self._data = data._data
+        # namedtuple in a list
+        elif (
+            len(data) > 0
+            and columns is None
+            and isinstance(data[0], tuple)
+            and hasattr(data[0], "_fields")
+        ):
+            # pandas behavior is to use the fields from the first tuple
+            # as the column names
+            columns = data[0]._fields
```

Comment:
This seems like a bug (or at least, potentially confusing behaviour). If I pass a list of dicts with disjoint keys I get a column index that is the union of the keys with nans in all the missing slots.

OTOH, if I pass a list of named tuples they are just treated like normal tuples except for picking out the names of the columns. So no disjointness is applied (and the keys are kind of "meaningless").

Reply:
Yeah this partially seems intentional when namedtuple support was implemented as it's documented in https://pandas.pydata.org/pandas-docs/stable/user_guide/dsintro.html#from-a-list-of-namedtuples